src/main/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,45 @@
+package stringcalculator;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+    private static final Pattern CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\n(.*)");
+
+    private static final String DEFAULT_VALUE = "0";
+    private static final String DELIMITER = ",|:";
+
+    private static final int DEFAULT_NUMBER = 0;
+    private static final int FIRST_MATCHED = 1;
+    private static final int SECOND_MATCHED = 2;
+
+    public int calculate(String expression) {
+        if (isNullAndEmpty(expression)) {
+            return DEFAULT_NUMBER;
+        }
+
+        String[] tokens = tokenize(expression);
+        CalculatorNumber result = new CalculatorNumber(DEFAULT_VALUE);
+
+        for (String token : tokens) {
+            CalculatorNumber calculatorNumber = new CalculatorNumber(token);
+            result = result.add(calculatorNumber);
+        }
+
+        return result.getNumber();
+    }
```

Comment:
```suggestion
    private Number result = new Number(DEFAULT_VALUE);

    public int calculate(String expression) {
        if (isNullAndEmpty(expression)) {
            return toInt(result);
        }
        return toInt(sum(toNumber(tokenize(expression))));
    }

    private Number sum(List<Number> numbers) {
        for(Number number: numbers) {
            result = result.add(number);
        }
        return result;
    }
```

메서드를 분리하여 의도를 분명히 드러내는 것은 어떨까요

> https://github.com/brainbackdoor/refactoring/blob/master/docs/Chapter_06_메서드_정리.md#extract-method-메서드-추출

Reply:
도전해보겠습니다 😄